From 787ecf28f3ff9caf04af4758910382bc7f4519cc Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Wed, 11 Apr 2018 22:06:13 +0200
Subject: [PATCH 2/4] Revert "updater: Fix and improve allowing devices to
 suppress BLKDISCARD"

The whole point of the flag, was to allow devices with slow secure
wiping to skip it. This is just making it even slower, thus defeating
the flags purpose. Also, this particular code has nothing to do with
regular wiping, but only executed during installing/updating.

This reverts commit 6f54468d5be103371ee7c249dbf4d8dad5713e54.

Change-Id: I5f910520b58ceb20037091bba8a762cca026c22b
---
 updater/blockimg.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/updater/blockimg.cpp b/updater/blockimg.cpp
index 9a0df627..c3891690 100644
--- a/updater/blockimg.cpp
+++ b/updater/blockimg.cpp
@@ -60,11 +60,7 @@
 // Set this to 0 to interpret 'erase' transfers to mean do a
 // BLKDISCARD ioctl (the normal behavior).  Set to 1 to interpret
 // erase to mean fill the region with zeroes.
-#ifdef SUPPRESS_EMMC_WIPE
-#define DEBUG_ERASE  1
-#else
 #define DEBUG_ERASE  0
-#endif
 
 static constexpr size_t BLOCKSIZE = 4096;
 static constexpr const char* STASH_DIRECTORY_BASE = "/cache/recovery";
@@ -1373,10 +1369,12 @@ static int PerformCommandErase(CommandParameters& params) {
       // length in bytes
       blocks[1] = (range.second - range.first) * static_cast<uint64_t>(BLOCKSIZE);
 
+#ifndef SUPPRESS_EMMC_WIPE
       if (ioctl(params.fd, BLKDISCARD, &blocks) == -1) {
         PLOG(ERROR) << "BLKDISCARD ioctl failed";
         return -1;
       }
+#endif
     }
   }
 
-- 
2.15.1

